kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+    \retval 2               Request completed successfully.
+    \retval 3               Stream type command is in progress.
+    \retval 4               GD syscalls are busy.
+
+    \sa syscall_gdrom_send_command(), syscall_gdrom_run_commands()
+*/
+int syscall_gdrom_check_command(uint32_t id, void *status);
+
+/** \brief   Process queued GDROM commands.
+    \ingroup system_calls
+
+    This function starts processing queued commands. This must be 
+    called a few times to process all commands.
+
+    \sa syscall_gdrom_send_command(), syscall_gdrom_check_command()
+*/
+void syscall_gdrom_run_commands(void);
+
+/** \brief   Abort a queued GDROM command.
+    \ingroup system_calls
+
+    This function tries to abort a previously queued command.
+
+    \param  id              The request id (>=0) to abort.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_abort_command(uint32_t id);
+
+/** \brief   Sets/gets the sector mode for read commands.
+    \ingroup system_calls
+
+    This function sets/gets the sector mode for read commands.
+
+    \param  status          The pointer to a struct of four 32 bit integers 
+                            containing new values, or to receive the old 
+                            values.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_gdrom_sector_mode(void *mode);
+
+
+/** \brief   Setup GDROM DMA callback
+    \ingroup system_calls
+
+    This function sets up DMA transfer end callback for 
+    CMD_DMAREAD_STREAM_EX (see cdrom.h).
+
+    \param  callback        The function to call upon completion of the DM.
+    \param  param           The data to pass to the callback function.
+*/
+void syscall_gdrom_dma_callback(uintptr_t callback, void *param);
+
+/** \brief   Initiates a GDROM DMA transfer
+    \ingroup system_calls
+
+    This function initiates a DMA transfer for 
+    CMD_DMAREAD_STREAM_EX (see cdrom.h).
+
+    \param  id              The request id (>=0).
+    \param  params          The pointer to two 32-bit integers. The first 
+                            element indicates the destination address, and 
+                            the second element identifies how many bytes to 
+                            transfer.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_dma_transfer(uint32_t id, int *params);
+
+/** \brief   Checks a GDROM DMA transfer
+    \ingroup system_calls
+
+    This function checks the progress of a DMA transfer for 
+    CMD_DMAREAD_STREAM_EX (see cdrom.h).
+
+    \param  id              The request id (>=0).
+    \param  size            The pointer to receive the remaining amount of
+                            bytes to transfer.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_gdrom_dma_check(uint32_t id, int *size);
```

Comment:
```suggestion
int syscall_gdrom_dma_check(uint32_t id, size_t *size);
```
If it's literally "bytes" or "size" or "count," that's a pretty good sign it should take a `size_t`.